src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -860,7 +868,6 @@ public void updateFromTopic(Topic topic) {
                     lastModifiedTime = Coerce.toLong(topic);
                     break;
                 case PERSISTED_CURRENT_PROCESSING_FILE_HASH:
-                    //TODO: the scenario of upgrading from older version needs to be handled in next PR.
                     fileHash = Coerce.toString(topic);
```

Comment:
By not calling the logic you added below here you are opening the door for this not setting the correct file hash. For example if on the previous file there was a `PERSISTED_CURRENT_PROCESSING_FILE_NAME` and no hash, when you call this it won't change that file name to a hash if (possible)

Reply:
The reason I put it out of the switch is that there is no guarantee for the order of the topic. 
1. If the fileName topic is called before the fileHash, then the embedded conversion works.
2. But if fileHash topic is called before the fileName, then there is no fileName to be used for locate the file.